### PR TITLE
Move PodPriorityResolution e2e to integration

### PR DESCRIPTION
**Pull request by damemi**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Following https://github.com/kubernetes/kubernetes/pull/80821, this moves the e2e test checking pod priority resolution to integration to free up e2e resources

```release-note
NONE
```
/sig scheduling

### CODE REVIEW

File: test/integration/scheduler/preemption_test.go
Comment (ravisantoshgudimetla) [inline test/integration/scheduler/preemption_test.go]: Can you check the priority value as well?

Comment (damemi) [reply test/integration/scheduler/preemption_test.go]: done

**Comment by alculquicondor** [comment]
It looks like this is flaky?

/hold

**Comment by alculquicondor** [comment]
Even if the test passes after retrying, I would be opposed to merge this. We shouldn't be introducing a test that is flaky.

**Comment by damemi** [comment]
@alculquicondor yeah, I just wanted to see if I could get a better idea of the flake after retrying. Do you have any suggestions to improve it? Since this is currently an e2e test, I'm not sure where the flakiness is introduced in moving it. We would like to move some of these e2e's to integration in this manner, so if I can learn how that should be done without introducing flakes it would be very helpful

**Comment by alculquicondor** [comment]
Looking at other tests, it looks like you have to create a node before running the pod.
You should be able to run integration tests locally fairly easy. Add `src/k8s.io/kubernetes/third_party/etcd` to your PATH

**Comment by damemi** [comment]
weird that this is failing in CI, because I added the change to create a node and it works fine locally:
```
$ make test-integration WHAT=./test/integration/scheduler KUBE_TEST_ARGS="-run ^TestPodPriorityResolution$"
+++ [0807 13:22:52] Checking etcd is on PATH
/home/mdame/go/src/k8s.io/kubernetes/third_party/etcd/etcd
+++ [0807 13:22:52] Starting etcd instance
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /tmp/tmp.mGY280RjIS --listen-client-urls http://127.0.0.1:2379 --debug > "/dev/null" 2>/dev/null
Waiting for etcd to come up.
+++ [0807 13:22:52] On try 1, etcd: : http://127.0.0.1:2379
{"action":"set","node":{"key":"/_test","value":"","modifiedIndex":4,"createdIndex":4}}
+++ [0807 13:22:52] Running integration test cases
Running tests for APIVersion: v1,admissionregistration.k8s.io/v1,admissionregistration.k8s.io/v1beta1,admission.k8s.io/v1beta1,apps/v1,apps/v1beta1,apps/v1beta2,auditregistration.k8s.io/v1alpha1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2beta1,autoscaling/v2beta2,batch/v1,batch/v1beta1,batch/v2alpha1,certificates.k8s.io/v1beta1,coordination.k8s.io/v1beta1,coordination.k8s.io/v1,extensions/v1beta1,events.k8s.io/v1beta1,imagepolicy.k8s.io/v1alpha1,networking.k8s.io/v1,networking.k8s.io/v1beta1,node.k8s.io/v1alpha1,node.k8s.io/v1beta1,policy/v1beta1,rbac.authorization.k8s.io/v1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,scheduling.k8s.io/v1alpha1,scheduling.k8s.io/v1beta1,scheduling.k8s.io/v1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,storage.k8s.io/v1alpha1,
+++ [0807 13:22:55] Running tests without code coverage
ok  	k8s.io/kubernetes/test/integration/scheduler	0.046s [no tests to run]
+++ [0807 13:23:05] Cleaning up etcd
+++ [0807 13:23:05] Integration test cleanup complete
```

**Review by ravisantoshgudimetla — COMMENTED** [review]
This LGTM

/cc @bsalamat

File: test/e2e/scheduling/preemption.go
Comment (Huang-Wei) [inline test/e2e/scheduling/preemption.go]: May I know why we should remove the e2e test? Due to portability?

**Comment by damemi** [comment]
Bumping, this is green in CI so wondering if there is any more feedback to it?

File: test/integration/scheduler/preemption_test.go
Comment (alculquicondor) [inline test/integration/scheduler/preemption_test.go]: use `t.Run`

File: test/e2e/scheduling/preemption.go
Comment (alculquicondor) [inline test/e2e/scheduling/preemption.go]: Move this comment to the new test.

File: test/integration/scheduler/preemption_test.go
Comment (alculquicondor) [inline test/integration/scheduler/preemption_test.go]: This check doesn't make much sense. `initPausePod` already sets the `Spec.Priority`. What's the output we expect from the scheduler? It looks like everything is already covered by `runPausePod`.

File: test/integration/scheduler/preemption_test.go
Comment (alculquicondor) [inline test/integration/scheduler/preemption_test.go]: Add a negative case? one for an unknown priority, and expect the scheduler to fail?

File: test/integration/scheduler/preemption_test.go
Comment (damemi) [reply test/integration/scheduler/preemption_test.go]: If I understand correctly, we are actually setting `PriorityClassName` (string) and verifying that it resolves to the expected `Priority` (int). So, part of the test is checking that `pod.Spec.Priority` is set by the priority admission plugin

File: test/integration/scheduler/preemption_test.go
Comment (damemi) [reply test/integration/scheduler/preemption_test.go]: Similar to my above reply, we're not setting integer values, we are setting a string and checking that it resolves to an integer, so I can't add a negative value check but I can add one for an unknown string and check that it resolves to nothing if you want. What do you think?

Comment (alculquicondor) [reply test/integration/scheduler/preemption_test.go]: Sorry, I didn't mean a negative number, just a negative case, i.e. a case that should be rejected.

File: test/integration/scheduler/preemption_test.go
Comment (alculquicondor) [reply test/integration/scheduler/preemption_test.go]: I see. Sounds good

File: test/integration/scheduler/preemption_test.go
Comment (damemi) [reply test/integration/scheduler/preemption_test.go]: Ah, I see then I misunderstood. I added a case for an invalid priority that should be rejected

**Comment by damemi** [comment]
@alculquicondor updated with your feedback, please let me know if there are any other changes you'd like

**Review by ravisantoshgudimetla — APPROVED** [review]
/approve

Thanks for the PR @damemi

**Pull request merged**
